### PR TITLE
Expose authority id for all authenticaiton methods

### PR DESCRIPTION
**Pull request by ohadbitt**
**Fixes:**
- Device authentication and certificate authentication did not expose a way to set authority id

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java [hunk 1]
```diff
@@ -147,17 +147,31 @@ public static ConnectionStringBuilder createWithAadApplicationCredentials(String
...
 
-    public static ConnectionStringBuilder createWithDeviceCodeCredentials(String resourceUri) {
+    public static ConnectionStringBuilder createWithDeviceCodeCredentials(String resourceUri, String authorityId) {
         if (StringUtils.isEmpty(resourceUri)) {
             throw new IllegalArgumentException("resourceUri cannot be null or empty");
         }
-        return new ConnectionStringBuilder(resourceUri);
+
+        ConnectionStringBuilder csb = new ConnectionStringBuilder(resourceUri);
+        csb.aadAuthorityId = authorityId;
```
Comment (vladikbr) [inline data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java:156]: What happens if authorityId is empty?
To what will it default?

File: data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java [hunk 2]
```diff
@@ -197,7 +197,7 @@ private AuthenticationResult acquireAccessTokenUsingDeviceCodeFlow() throws Exce
 
     private AuthenticationResult waitAndAcquireTokenByDeviceCode(DeviceCode deviceCode, AuthenticationContext context)
             throws InterruptedException {
-        int timeout = 20 * 1000;
+        int timeout = 60 * 1000;
```
Comment (vladikbr) [inline data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java:200]: I think it's a bit too much

**Review by vladikbr — CHANGES_REQUESTED** [review]
Pls see comments

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java (hunk 1 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java:156]: default is microsoft.com

File: data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java (hunk 2 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java:200]: it is a race to make it work - its not the first time i see people struggle to make it in 20 seconds ..  
how much do you think

Comment (vladikbr) [reply data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java:200]: Fair enough

File: data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java (hunk 1 above)
Comment (vladikbr) [reply data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java:156]: "common" is better, but check with Ran that it is correct and check that it works for you correctly

**Pull request merged**